corehq/apps/domain/views/base.py

```diff
@@ -157,3 +154,15 @@ def main_context(self):
     def page_url(self):
         if self.urlname:
             return reverse(self.urlname, args=[self.domain])
+
+
+def redirect_to_domain(request):
+    """Allows us to use eg commcarehq.org/a/DOMAIN/settings/users/web
+    in documentation and have it redirect to the user's domain"""
+    # switch out the DOMAIN placeholder so it doesn't match this view
+    resolvable_path = '/a/example/' + request.path.removeprefix('/a/DOMAIN/')
```

Comment:
Any URL beginning with `/a/DOMAIN/` resolves to this view, so this is attempting to see what it'd resolve to with a different domain name.  The actual domain used doesn't matter, because it gets replaced in `get_url` anyways.

I'd initially done this because the `select` view needed a `next_view` name, but I suppose now that I switched it to taking a `get_url` callable, I could in theory just change this function to
```python
def redirect_to_domain(request):
    def get_url(domain):
         return f'/a/{domain}/' + request.path.removeprefix('/a/DOMAIN/')
    return select(request, always_show_list=True, get_url=get_url)
``` 

However, Clayton brought up on slack the point that we should be cautious about which links can be redirected in this way, and that approach is very blunt.  Rather, I've been meaning to rework this a little to make a whitelist of allowable URL kwargs - domain and report slugs for sure, possibly others.  This PR is kinda sitting until I get around to that though.